### DISCUSSION TITLE
Using Projen CDK app

### DISCUSSION THREAD

**Discussion by shawnwang-avl** [projen/projen#2060] [General]
Our service is a TypeScript express server built with NPM. We use Github Actions to build Docker image, publish to ECR, track by CodePipeline, produce image artifact with CodeBuild, and finally deploy with App Runner.

How to best integrate a projen CDK app with our non-projen build system? Do we have to use YARN?
Should we still follow AWS's recommendation of putting CDK infrastructure code along with service source code?

**Comment by mrgrain** [comment]
You can use any supported packages manager: `yarn` `npm` `pnpm`
There is an option on the new command for this or see the API docs for `packageManager`.

--- 

I'm not aware of docs around migrating an existing project. My recommendation is to create a new projen app and copy your code into it.

**Reply by ahammond** [reply]
Be advised that `pnpm` doesn't appear to work with `cdk-pipelines`. https://github.com/aws/aws-cdk/issues/25820
